lib/specinfra/command/debian/base/package.rb

```diff
@@ -22,7 +22,7 @@ def install(package, version=nil, option='')
     end
 
     def get_version(package, opts=nil)
-      "dpkg-query -f '${Status} ${Version}' -W #{package} | sed -n 's/^install ok installed //p'"
+      "dpkg-query -f '${Status} ${Version}' -W #{package} | sed -n 's/^install ok installed \|hold ok installed //p'"
```

Comment:
Ah, sorry, one more fix. `\|hold` should be `\\|hold` .

Reply:
Not sure that this is needed.

For example when package `docker-ce` is held:

```shell
▶ dpkg-query -f '${Status} ${Version}' -W docker-ce                                                           
hold ok installed 5:18.09.0~3-0~ubuntu-bionic
```

It is with `\|`

```shell
▶ dpkg-query -f '${Status} ${Version}' -W docker-ce | sed -n 's/^install ok installed \|hold ok installed //p'
5:18.09.0~3-0~ubuntu-bionic
```

It is with `\\|`

```shell
▶ dpkg-query -f '${Status} ${Version}' -W docker-ce | sed -n 's/^install ok installed \\|hold ok installed //p'
(EMPTY)
```

For example when package `docker-ce` is not held:

```shell
▶ dpkg-query -f '${Status} ${Version}' -W docker-ce                                                           
install ok installed 5:18.09.0~3-0~ubuntu-bionic
```

It is with `\|`

```shell
▶ dpkg-query -f '${Status} ${Version}' -W docker-ce | sed -n 's/^install ok installed \|hold ok installed //p' 
5:18.09.0~3-0~ubuntu-bionic
```

It is with `\\|`

```shell
▶ dpkg-query -f '${Status} ${Version}' -W docker-ce | sed -n 's/^install ok installed \\|hold ok installed //p'
```

Reply:
`\` disappears when it's between double quotes in the Ruby code.

Reply:
It's clear to not work properly if you run this code. So you did not run your code before you pushed this PR. I won't accept this kind of PR.

Reply:
@mizzy While I understand your PR discipline, note that running the integration suite is another topic. Since I've personally made mistakes while translating the code from a private code base (by monkey-patching) I think it's understandable. 

What's most important here is to get this fixed upstream :)

Reply:
Followed this up in #672 . Just my two cents, I think we can reach to an agreement before rushing to close PRs.